source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -3354,6 +3358,23 @@ void Graph::propertyEditor()
                 std::string name = _currUiNode->getNodeGraph()->getParent()->createValidChildName(temp);
                 _currUiNode->getNodeGraph()->setName(name);
                 _currUiNode->setName(name);
+
+                for (UiNodePtr node : _graphNodes)
+                {
+                    std::vector<UiPinPtr> inputs = node->inputPins;
+                    if (node->getInput() == nullptr)
+                    {
+                        for (size_t i = 0; i < inputs.size(); i++)
+                        {
+                            std::string inputName = inputs[i]->_name;
+                            UiNodePtr inputNode = node->getConnectedNode(inputName);
+                            if (inputNode != nullptr && inputNode->getName() == name && node->getNode())
+                            {
+                                node->getNode()->getInput(inputName)->setAttribute("nodegraph", name);
+                            }
+                        }
+                    }
+                }
```

Comment:
I copy pasted some other places's code to here since I'm new to the code base, not quite sure how to do this update elegantly. I appreciate your futther suggestion!